src/sinks/file/mod.rs

```diff
@@ -181,12 +180,16 @@ impl FileSink {
             .expect("unable to compute next deadline")
     }
 
-    async fn run(&mut self, input: impl Stream<Item = Event> + Send + Sync) -> crate::Result<()> {
-        pin_mut!(input);
+    async fn run(&mut self, mut input: BoxStream<'_, Result<Event, ()>>) -> crate::Result<()> {
         loop {
             tokio::select! {
                 event = input.next() => {
                     match event {
+                        Some(Err(())) => {},
+                        Some(Ok(event)) => {
+                            self.process_event(event).await;
+                            self.acker.ack(1);
```

Comment:
We can't ack at this level, as this would ignore the error conditions.

I think if we make the `process_event` return `bool`, or `Result<(), ()>` it would be possible to lift this up. Better yet: return an `enum ShouldAck { Ack, NoAck }` fore more explicitness.

Alternatively, we could move the logic of issuing the `self.acker.ack(1);` down the line - but it might hurt the clarity of the code. The return type approach would guarantee a signle pass to the caller (and calling the `self.acker.ack(1);`).

Reply:
I think it is. It was wrong before as well though.
This implementation would ack even if there was an error. I might be getting the concept of our acks wrong though...

Reply:
We have to ack to continue to make progress, so it's more a way of saying "I've done all I'm going to do with this event" than "this was processed successfully". I'd like to make that better in the future but for now I think we can just match the existing behavior.